### PR TITLE
Fix Android app crash when app restart and improve RouteExample

### PR DESCRIPTION
**Pull request by thongdoan**
- Fix Android app crash when app restart
- Improve RouteExample.Droid (show result on screen)

### CODE REVIEW

File: TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs [hunk 1]
```diff
@@ -20,7 +21,12 @@ public MainViewModel(IMvxNavigationService routingService)
             _routingService = routingService;
         }
 
-        private IMvxCommand _showACommand;
+		public MainViewModel()
+		{
+			_routingService = Mvx.Resolve<IMvxNavigationService>();
```
Comment (martijn00) [inline TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs]: Can you use constructor injection instead?

File: TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs [hunk 2]
```diff
@@ -62,5 +68,19 @@ public IMvxCommand ShowRandomCommand
...
         }
+
+		
+		private string _lastResult;
+		public string LastResult {
+			get {
+				return _lastResult;
+			}
+			set {
+				if (object.Equals(_lastResult, value))
```
Comment (martijn00) [inline TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs]: Use `SetProperty()` of MvvmCross to do this.

File: TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs [hunk 3]
```diff
@@ -45,7 +51,7 @@ public IMvxCommand ShowBCommand
                 {
                     //var result = await _routingService.Navigate<User, User>("mvx://test/?id=" + Guid.NewGuid().ToString("N"), new User("MvvmCross2", "Test2"));
                     var result = await _routingService.Navigate<TestBViewModel, User, User>(new User("MvvmCross", "Test"));
-                    var test = result?.FirstName;
+					this.LastResult = result?.FirstName;
```
Comment (martijn00) [inline TestProjects/Navigation/RoutingExample.Core/ViewModels/MainViewModel.cs]: We don't use `this` in our code base.

File: MvvmCross/Droid/Droid/Views/MvxActivity.cs [hunk 4]
```diff
@@ -66,7 +66,7 @@ public override void SetContentView(int layoutResId)
             onGlobalLayout = (sender, args) =>
             {
                 view.ViewTreeObserver.GlobalLayout -= onGlobalLayout;
-                ViewModel.Appeared();
+                ViewModel?.Appeared();
```
Comment (martijn00) [inline MvvmCross/Droid/Droid/Views/MvxActivity.cs]: We might need to add those null checks on iOS and Windows as well.

**Comment by martijn00** [comment]
In general also using spaces instead of tabs for indentation.

**Comment by thongdoan** [comment]
Yes, I forget delete default constructor
About indentation, it is default behavior of Xamarin Studio

**Comment by kjeremy** [comment]
@thongdoan Why did you close this?

**Comment by thongdoan** [comment]
Sorry, I don't have more experience, I just delete my fork, I just create new fork, then I will make another pull request

**Comment by kjeremy** [comment]
Oh you don't have to do that :)

You can make the changes as requested in the review and keep pushing them until we're happy with the code.  Then to split your PR back into those two commits you can use `git rebase -i` to reorder and squash.  Then you can force push the branch (`git push -f`) to overwrite the existing commits with the new ones.

**Comment by thongdoan** [comment]
yes, thanks, you save my life, i will do it in future :)

**Pull request closed without merging**